### ISSUE TITLE
Azure File Storage

### ISSUE THREAD

**Issue by JoshTango** [OrchardCMS/OrchardCore#4431]
I would like to use Orchard Core as a mult-tennant application and I want to save SQLite files to drive for each tennant.  I would prefer Orchard Core save these and access them in Azure File Storage.  That way if I use the docker container the container does not fill up with all the database files.  The same if I use a VM.  Does Orchard Core support this ability and if so is there guidance or documentation somewhere?  Also I noticed you published a docker container recently.  Will there be a way to configure that container to save the database files externally such as with Axure File Storage?

**Comment by carlwoodhouse** [comment]
In terms fo your db; In azure docker containers you can mount a storage fileshare at any given path (https://docs.microsoft.com/en-us/azure/app-service/containers/how-to-serve-content-from-azure-storage) , it's possible to set the path of appdata in your container using `ORCHARD_APP_DATA` env variable.

If that helps ...

that been said id seriously consider the limitations of sqllite and look at sql azure instead ..

**Comment by JoshTango** [comment]
Well what if I expect thousands of people to sign up and create a 1000 little SQLite databases files?  Orchard core can spawn multiple threads and talk to several SQLite files in parallel can it not?  And with caching turned on why should I care about the limitations of SQLite?

**Comment by carlwoodhouse** [comment]
For sure :) just a suggestion; i might be worried about the concurrent
write limitations of those tenants if the users of them are out of your
control! but by all means it would work :)

**Comment by JoshTango** [comment]
The link you provided was helpful

**Comment by JoshTango** [comment]
If I were to get a managed sql instance I could then create thousands of tenants databases inside of  that yes?  that's what you mean?  The cost is not much more than File Storage it seems.

**Comment by carlwoodhouse** [comment]
Yup :) plus you get all the benefits of sql azure; automated backup; etc
